libvast/vast/system/shutdown.hpp

```diff
@@ -29,26 +32,58 @@ struct parallel;
 
 namespace vast::system {
 
-/// Performs an asynchronous shutdown of a set of actors, followed by
-/// terminating the actor in the calling context. The shutdown process runs
-/// either sequentially or in parallel. As soon as all actors have terminated,
-/// the calling actor exits. The shutdown process involves sending an EXIT
-/// message with reason `user_shutdown`.
+/// Performs an asynchronous shutdown of a set of actors, or terminates the
+/// current process if that is not possible. The shutdown process runs
+/// either sequentially or in parallel, based on the provided policy parameter.
+/// This involves monitoring the actor, sending an EXIT message with reason
+/// `user_shutdown`, and then waiting for the DOWN. As soon as all actors have
+/// terminated, the calling actor exits with `caf::exit_reason::user_shutdown`.
+/// If an actor does not respond with a DOWN within the provided grace period,
+/// we send out another EXIT message with reason `kill`. If the actor still
+/// does not terminate within the provided timeout, the process aborts hard. If
+/// these failure semantics do not suit your use case, consider using the
+/// function `terminate`, which allows for more detailed control over the
+/// shutdown sequence.
 /// @param self The actor to terminate.
 /// @param xs Actors that need to shutdown before *self* quits.
+/// @param grace_period The amount of time to wait until all actors terminated
+///        cleanly.
+/// @param kill_timeout The timeout befor giving and calling `abort(3)`.
```

Comment:
s/befor giving/before giving up/

Also, we should probably mention that this timeout starts *after* the grace period has ended.